splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -1190,6 +1189,7 @@ private static void snapshot(ResultSet rs, String snapshotName, String schemaNam
                     new SnapshotDescriptor(snapshotName, schemaName, objectName, conglomerateNumber,creationTime, null);
             admin.snapshot(sname, "splice:" + conglomerateNumber);
             dd.addSnapshot(descriptor, tc);
+            snapshotList.add(sname);
```

Comment:
My understanding is that if "admin.snapshot()" or "dd.addSnapshot()" fails, we should abort/return early and we won't reach the line "snapshotList.add(sname)". Later on, deleteSnapshots() in the below code won't delete a snapshot that fails to be created:

        try {
            dd.startWriting(lcc);

            ResultSet rs = getResultSet(sql1, schemaName, tableName);
            snapshot(rs, snapshotName, schemaName, dd, tc, snapshotList);

            rs = getResultSet(sql2, schemaName, tableName);
            snapshot(rs, snapshotName, schemaName, dd, tc, snapshotList);
        }
        catch (Exception e)
        {
            deleteSnapshots(snapshotList);
            throw e;
        }